### DISCUSSION TITLE
What's the Correct Batch Size in Distributed Data Parallel Training?

### DISCUSSION THREAD

**Discussion by zhenhuahu** [Lightning-AI/pytorch-lightning#2684] [DDP / multi-GPU / multi-node] [answered]
### Before asking:   
1. search the issues.   
2. search the docs.    

#### train_loader = DataLoader(dataset=train_dataset, batch_size=2, shuffle=True, num_workers=4, pin_memory=True, drop_last=True)
trainer = pl.Trainer(gpus=8, num_nodes = 4 distributed_backend='ddp')
trainer.fit(model, train_dataloader=train_loader)
   

#### What's your environment?

 - OS: [e.g. iOS, Linux, Win]
 - Packaging [e.g. pip, conda]
 - Version [e.g. 0.5.2.1]

**Comment by rohitgr7** [comment, marked as answer]
Is this what you need? https://pytorch-lightning.readthedocs.io/en/latest/multi_gpu.html#batch-size

**Comment by zhenhuahu** [comment]
> Is this what you need? https://pytorch-lightning.readthedocs.io/en/latest/multi_gpu.html#batch-size

Yes. That's exactly what I want. Thank you so much!
